File: DesignPatternsDotNet/ChainOfResponsibility/RealHandlerTwo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ChainOfResponsibility
{
    public class RealHandlerTwo : MyHandler
    {
        public override void DoThis(Task task)
        {
            if (task.Value == 0)
            {
                Console.WriteLine("This is Handler Two and value = " + task.Value.ToString());
            }
            else
            {
                myNextHandler.DoThis(task);
            }
        }
    }
}

File: DesignPatternsJava/src/main/java/com/ChainOfResponsibility/RealHandlerTwo.java
package com.ChainOfResponsibility;

public class RealHandlerTwo extends MyHandler {

	public void doThis(Task task){
		if(task.getValue()==0)
		{
			System.out.println("This is Handler two and value = "+ task.getValue());			
		}
		else{
			myNextHandler.doThis(task);	
		}		
	}
	public RealHandlerTwo() {		
	}

}
